src/worker/ingestion/process-event.ts

```diff
@@ -472,14 +493,17 @@ export class EventsProcessor {
 
         if (properties['$set'] || properties['$set_once'] || properties['$increment']) {
             const filteredIncrementProperties = filterIncrementProperties(properties['$increment'])
-
-            await this.updatePersonProperties(
+            const updatedProperties = await this.updatePersonProperties(
                 teamId,
                 distinctId,
                 properties['$set'] || {},
                 properties['$set_once'] || {},
                 filteredIncrementProperties
             )
+
+            delete properties['$set']
+            delete properties['$set_once']
+            properties = { ...properties, ...updatedProperties }
```

Comment:
I'd also add a code comment here, that `$set` and `$set_once` come back inside `updatedProperties`, otherwise it reads like they are removed from being stored on an event.

Alternatively, rename `updatedProperties` to `updatedSetAndSetOnce` and let the code speak for itself 🎶